src/types/floatob.rs

```diff
@@ -29,7 +30,23 @@ impl PyFloat {
 
     /// Gets the value of this float.
     pub fn value(&self) -> c_double {
-        unsafe { ffi::PyFloat_AsDouble(self.as_ptr()) }
+        #[cfg(not(Py_LIMITED_API))]
+        {
+            return self.as_double_fastpath();
+        }
+
+        #[cfg(Py_LIMITED_API)]
+        unsafe {
+            return ffi::PyFloat_AsDouble(self.as_ptr());
+        }
+    }
+
+    // private safe wrapper around PyFloat_AS_DOUBLE
+    // not exposed because .value() uses this on the unlimited API anyway
+    #[cfg(not(Py_LIMITED_API))]
+    fn as_double_fastpath(&self) -> c_double {
```

Comment:
Couldn't `extract` just call `value` directly, avoid the need for this additional helper method?

Reply:
I started out like that but noted it's slower to do the downcast and then call `.value()` for `abi3` (the check is redundant on abi3). So I decided to use a fastpath method so we can't accidentally remove the `#[cfg]` from the branch in extract and deoptimize `abi3`.

Reply:
My thinking was to still start with `downcast_exact` and only for `not(Py_LIMITED_API)` and then call into `.value()` which should expand only to the `not(Py_LIMITPED_API)` code path. Hence the redundant call to `PyFloat_AsDouble` would never be emitted.

So basically, literally replacing `.as_double_fastpath()` by `.value()` should yield the exact same generated code, should it not?

Reply:
Done 👍